src/openrct2/world/Wall.cpp

```diff
@@ -35,26 +35,24 @@
  */
 void wall_remove_at(const CoordsXYRangedZ& wallPos)
 {
-    TileElement* tileElement;
-
-repeat:
-    tileElement = map_get_first_element_at(wallPos);
-    if (tileElement == nullptr)
-        return;
-    do
+    for (TileElement* tileElement = map_get_first_element_at(wallPos); tileElement;
+         tileElement = map_get_first_element_at(wallPos))
     {
-        if (tileElement->GetType() != TILE_ELEMENT_TYPE_WALL)
-            continue;
-        if (wallPos.baseZ >= tileElement->GetClearanceZ())
-            continue;
-        if (wallPos.clearanceZ <= tileElement->GetBaseZ())
-            continue;
-
-        tile_element_remove_banner_entry(tileElement);
-        map_invalidate_tile_zoom1({ wallPos, tileElement->GetBaseZ(), tileElement->GetBaseZ() + 72 });
-        tile_element_remove(tileElement);
-        goto repeat;
-    } while (!(tileElement++)->IsLastForTile());
+        do
+        {
+            if (tileElement->GetType() != TILE_ELEMENT_TYPE_WALL)
+                continue;
+            if (wallPos.baseZ >= tileElement->GetClearanceZ())
+                continue;
+            if (wallPos.clearanceZ <= tileElement->GetBaseZ())
+                continue;
+
+            tile_element_remove_banner_entry(tileElement);
+            map_invalidate_tile_zoom1({ wallPos, tileElement->GetBaseZ(), tileElement->GetBaseZ() + 72 });
+            tile_element_remove(tileElement);
+            break;
+        } while (!(tileElement++)->IsLastForTile());
```

Comment:
Having a `break` just before the `while` seems weird. All it is doing at the beginning is validating whether it's ok to remove and if it is, it proceeds and exits. This is what I come up with, I'll leave it here for your consideration and other people in the team too
```c++
    bool removalDone = false;
    while (!removalDone)
    {
        if (tileElement->GetType() == TILE_ELEMENT_TYPE_WALL && wallPos.baseZ < tileElement->GetClearanceZ()
            && wallPos.clearanceZ > tileElement->GetBaseZ())
        {
            tile_element_remove_banner_entry(tileElement);
            map_invalidate_tile_zoom1({ wallPos, tileElement->GetBaseZ(), tileElement->GetBaseZ() + 72 });
            tile_element_remove(tileElement);
            removalDone = true;
        }
        else
        {
            removalDone = (tileElement++)->IsLastForTile();
        }
    }
```

Reply:
In general all of the tile functions are do while loops with a break like that.

Reply:
It seems way better like this though.